### PR TITLE
[Refinery] Create new Integer Group and move integer relvant constraint to this new group

### PR DESCRIPTION
**Pull request by legionth**
As addition to #1707 . We said we would open PR to handle the move of `Validation` and `Transformation` to the top level.

In discussion with @mjansenDatabay  we came up with this solution:
* Creating a new Group based on https://raw.githubusercontent.com/ILIAS-eLearning/ILIAS/trunk/docs/documentation/input-processing.md (In this case the `Integer` group)
* Matching `Constraints` or `Transformations` will be moved to this new group and the origin methods will be removed from `Validation\Factory`
* Adapt Namespace and move tests to new namespace locations.

@klees could you give us feedback if this is the intended approach? I would open dedicated PRs for the new groups (`String` etc.) and move the matching `Constraints` to this new group.

### CODE REVIEW

File: src/Refinery/Factory.php [hunk 1]
```diff
@@ -20,12 +20,19 @@ class Factory
...
+
 	/**
 	 * @param \ILIAS\Data\Factory $dataFactory
+	 * @param \ilLanguage $language
 	 */
-	public function __construct(\ILIAS\Data\Factory $dataFactory)
+	public function __construct(\ILIAS\Data\Factory $dataFactory, \ilLanguage $language)
 	{
 		$this->dataFactory = $dataFactory;
+		$this->lanugage = $language;
```
Comment (mjansenDatabay) [inline src/Refinery/Factory.php]: `language`

File: src/Refinery/Factory.php [hunk 2]
```diff
@@ -20,12 +20,19 @@ class Factory
 	 */
 	private $dataFactory;
 
+	/**
+	 * @var \ilLanguage
+	 */
+	private $lanugage;
```
Comment (mjansenDatabay) [inline src/Refinery/Factory.php]: `language`

**Comment by mjansenDatabay** [comment]
@legionth Thx for the PR. IMO this is the right way we should go/continue to get rid of the `Contstraint` and `Transformation` classes/factories.

File: src/Refinery/Integer/Group.php [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+	 */
+	private $language;
+
+	public function __construct(Factory $dataFactory, \ilLanguage $language)
+	{
+		$this->dataFactory = $dataFactory;
+		$this->language = $language;
+	}
+
+	public function isGreaterThan(int $minimum) : GreaterThan
```
Comment (mjansenDatabay) [inline src/Refinery/Integer/Group.php:38]: Please add a short description here. Of course the method name is verbose in this case, but there might be future transformations which need more explanations in the docs.

File: src/Refinery/Integer/Group.php [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+	{
+		return new GreaterThan($minimum, $this->dataFactory, $this->language);
+	}
+
+	public function isLessThan(int $maximum) : LessThan
```
Comment (mjansenDatabay) [inline src/Refinery/Integer/Group.php:50]: Same here

**Review by klees — APPROVED** [review]
:+1: 

Thx a lot!

**Pull request merged**